reporter/src/main/kotlin/reporters/NoticeReporter.kt

```diff
@@ -138,13 +138,13 @@ class NoticeReporter : Reporter() {
         buildString {
             append(noticeReport.headers.joinToString(NOTICE_SEPARATOR))
 
-            val mergedFindings = noticeReport.findings.values.reduce { left, right ->
+            val mergedFindings = noticeReport.findings.values.takeIf { it.isNotEmpty() }?.reduce { left, right ->
                 left.apply {
                     right.forEach { (license, copyrights) ->
                         getOrPut(license) { mutableSetOf() } += copyrights
                     }
                 }
-            }.removeGarbage(copyrightGarbage).processStatements()
+            }?.removeGarbage(copyrightGarbage)?.processStatements() ?: sortedMapOf()
```

Comment:
147: How about replacing all three `?` along with `: sortedMapOf()` with a single `.orEmpty()`?

Reply:
I cannot use `orEmpty()` because the result must be a `SortedMap` (although that would be another thing to refactor, as sorting could happen later on).